hw02-maven/src/main/java/com/remifo/DIYArrayList.java

```diff
@@ -0,0 +1,313 @@
+package com.remifo;
+
+import java.lang.reflect.Array;
+import java.util.*;
+
+public class DIYArrayList<T> implements List<T> {
+    private Object[] array;
+    private int size;
+
+    public DIYArrayList() {
+        this.array = new Object[0];
+        size = 0;
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        for (Object t : array) {
+            if (t.equals(o)) return true;
+        }
+        return false;
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return null;
+    }
+
+    @Override
+    public Object[] toArray() {
+        return array;
+    }
+
+    @Override
+    public <T1> T1[] toArray(T1[] a) {
+        a = (T1[]) array;
+        return a;
+    }
+
+    @Override
+    public boolean add(T t) {
+        Object[] new_array = Arrays.copyOf(array, ++size);
```

Comment:
Принято писать в camelCase
```suggestion
        Object[] newArray = Arrays.copyOf(array, ++size);
```

Также увеличивать размер массива при каждом add() не рационально. Лучше увеличивать на большее количество элементов, но использовать только нужное количество.